networkit/cpp/graph/test/GraphToolsGTest.cpp

```diff
@@ -1039,4 +1032,35 @@ TEST_P(GraphToolsGTest, testEdgesSortedByWeight) {
         EXPECT_TRUE(G.checkConsistency());
     }
 }
+
+TEST_P(GraphToolsGTest, testEdgesRandomizer) {
+    if (!weighted())
+        return;
+    Aux::Random::setSeed(1, true);
+    ErdosRenyiGenerator generator(3000, 0.33, directed());
+    Graph G1 = generator.generate();
+    Aux::Random::setSeed(1, true);
+    Graph G(G1, true, directed());
+    GraphTools::randomizeWeights(G);
+
+    count edgesChecked = 0;
+    edgeweight average = 0.0;
+    edgeweight squareDiffSum = 0.0;
+    G.forEdges([&](node, node, edgeweight w) {
+        edgesChecked++;
+        edgeweight diff = (w - average);
+        average += diff / edgesChecked;
```

Comment:
This does not seem an average. Could you add some explanation about what are you computing here?

Reply:
it is th online algorithm for the average, it is indeed the average. `avg = avg + (newWeight - avg)/samples`;

Reply:
Why is an online algorithm necessary? Looks like this leads to more arithmetic operations than a simple sum and one final division.